mega-linter-runner/lib/upgrade.js

```diff
@@ -226,10 +226,10 @@ jobs:
       // V5 to V6 migration rules
       // GitHub actions
       {
-        regex: /actions\/checkout@v2/gm,
-        replacement: "actions/checkout@v3",
-        test: "uses: actions/checkout@v2",
-        testRes: "uses: actions/checkout@v3",
+        regex: /actions\/checkout@v3/gm,
+        replacement: "actions/checkout@v4",
+        test: "uses: actions/checkout@v3",
+        testRes: "uses: actions/checkout@v4",
```

Comment:
Here are some regexes to upgrade old configs. It should probably be a new section that takes @v3 to replace with @v4, so that the upgrade still works with the config generated at that point in time.

If you would be able to test a really old upgrade, probably upgrading from `actions/checkout@v2` directly to `actions/checkout@v4` could be correct, but it is needed to test that (ie, if the upgrade changed from one version of megalinter like Megalinter 5 to 6, then we have an upgrade from Megalinter 6 to 7, changing something in between might break the upgrade chain).

Reply:
Agreed :)
We need to be able to upgrade v2 to v3 then v3 to v4

A ne rule can be added, as they are processed iterativement, the script will update in the same run from v2 to v3 then to v3 to v4